framework/src/preconditioners/FieldSplitPreconditioner.C

```diff
@@ -60,22 +53,24 @@ FieldSplitPreconditioner::FieldSplitPreconditioner(const InputParameters & param
   // it is recommended to have a full Jacobian for using
   // the fieldSplit preconditioner
   bool full = getParam<bool>("full");
+
   // how variables couple
   std::unique_ptr<CouplingMatrix> cm = std::make_unique<CouplingMatrix>(n_vars);
   if (!full)
   {
+    const auto off_diag_rows = getParam<std::vector<NonlinearVariableName>>("off_diag_row");
+    const auto off_diag_columns = getParam<std::vector<NonlinearVariableName>>("off_diag_column");
+
     // put 1s on diagonal
     for (unsigned int i = 0; i < n_vars; i++)
       (*cm)(i, i) = 1;
 
     // off-diagonal entries
     std::vector<std::vector<unsigned int>> off_diag(n_vars);
-    for (unsigned int i = 0; i < getParam<std::vector<std::string>>("off_diag_row").size(); i++)
+    for (unsigned int i = 0; i < off_diag_rows.size(); i++)
```

Comment:
might as well update this to
```suggestion
    for (const auto i: index_range(off_diag_rows))
```